workflows/nanoseq.nf

```diff
@@ -0,0 +1,459 @@
+          ch_software_versions = ch_software_versions.mix(ALIGN_GRAPHMAP2.out.samtools_version.first().ifEmpty(null))
+          ch_samtools_multiqc  = ALIGN_GRAPHMAP2.out.ch_sortbam_stats_multiqc.ifEmpty([])
+       }
+
+       ch_bedtools_version = Channel.empty()
+       if (!params.skip_bigwig){
+
+          /*
+           * SUBWORKFLOW: Convert BAM -> BEDGraph -> BigWig
+           */
+          BEDTOOLS_UCSC_BIGWIG ( ch_view_sortbam )
+          ch_bedtools_version = ch_bedtools_version.mix(BEDTOOLS_UCSC_BIGWIG.out.bedtools_version.first().ifEmpty(null))
+          ch_software_versions = ch_software_versions.mix(BEDTOOLS_UCSC_BIGWIG.out.bedgraphtobigwig_version.first().ifEmpty(null))
+       }
+       if (!params.skip_bigbed){
+
+          /*
+           * SUBWORKFLOW: Convert BAM -> BED12 -> BigBED
+           */
+          BEDTOOLS_UCSC_BIGBED ( ch_view_sortbam )
+          ch_bedtools_version = ch_bedtools_version.mix(BEDTOOLS_UCSC_BIGBED.out.bedtools_version.first().ifEmpty(null))
+          ch_software_versions = ch_software_versions.mix(BEDTOOLS_UCSC_BIGBED.out.bed12tobigbed_version.first().ifEmpty(null))
+       }
+       ch_software_versions = ch_software_versions.mix(ch_bedtools_version.first().ifEmpty(null))
+
+       ch_view_sortbam
+          .map { it -> [ it[0], it[3] ] }
+          .set { ch_sortbam }
+    } else {
+       ch_sample
+           .map { it -> if (it[6].toString().endsWith('.bam')) [ it[0], it[6] ] }
+           .set { ch_sample_bam }
+       BAM_RENAME ( ch_sample_bam )
+       ch_sortbam = BAM_RENAME.out.sortbam_quant
+    }
+
+    ch_featurecounts_gene_multiqc       = Channel.empty()
+    ch_featurecounts_transcript_multiqc = Channel.empty()
+    if (!params.skip_quantification && (params.protocol == 'cDNA' || params.protocol == 'directRNA')) {
+
+       // Check that reference genome and annotation are the same for all samples if perfoming quantification
+       // Check if we have replicates and multiple conditions in the input samplesheet
+       REPLICATES_EXIST    = false
+       MULTIPLE_CONDITIONS = false
+       ch_sample.map{ it[2] }.unique().toList().set { fastas }
+       ch_sample.map{ it[3] }.unique().toList().set { gtfs }
+     // BUG: ".val" halts the pipeline ///////////////////////
+     //  if ( gtfs.map{it[0]} == false || fastas.map{it[0]} == false || gtfs.size().val != 1 || fasta.size().val != 1 ) {
+     //      exit 1, """Quantification can only be performed if all samples in the samplesheet have the same reference fasta and GTF file."
+     //              Please specify the '--skip_quantification' parameter if you wish to skip these steps."""
+     //  }
+     //  REPLICATES_EXIST    = ch_sample.map { it -> it[0].split('_')[-1].replaceAll('R','').toInteger() }.max().val > 1
+     //  MULTIPLE_CONDITIONS = ch_sample.map { it -> it[0].split('_')[0..-2].join('_') }.unique().count().val > 1

```

Comment:
Are these comments still needed or can they be deleted?

Reply:
I think I will keep this comment for a little longer after the transcript quantification part of the pipeline is updated in a later release